### PR TITLE
Task 2

### PR DESCRIPTION
**Pull request by oDoLaReKo**
I think that it shoud be ok now. I was very confused with those commit's

### CODE REVIEW

File: src/components/SquareBox/SquareBox.tsx [hunk 1]
```diff
@@ -28,30 +30,44 @@ const styles = StyleSheet.create({
...
+                ...squares,{color:randomElement(Object.values(COLORS)),shape:'circle'}
+            ]           
             ))
-        }, []);
+        } 
+
+const deleteSquare=()=>{   
+    squares.pop();
+    setSquares(squares=>[...squares]);    
+};
```
Comment (kaz231) [inline src/components/SquareBox/SquareBox.tsx]: Format this code pls

**Comment by oDoLaReKo** [comment]
3 errors:
1. There is problem with id that is assigned but never used idk why becouse it is used in 70 line " <Square key={id} color={square.color} /> : square.shape === 'circle'"
2. square is already declared in upper scope i read about it and this is a bug that we can changed by adding 
"no-shadow": "off",
"@typescript-eslint/no-shadow": ["error"] 
but I'll wait for your opinion
3. do not nest ternary expression i do not understand why

File: src/components/SquareBox/SquareBox.tsx [hunk 2]
```diff
@@ -1,57 +1,75 @@
...
-        <div className={css(styles.squares)}>
-            {squares.map((square) => <Square key={id} color={square} />)}
-        </div>
-    </div>;
-}
\ No newline at end of file
+      <div className={css(styles.squares)}>
+        {squares.map((square: Shape, id) => (square.shape === 'square'
+          ? <Square key={id} color={square.color} /> : square.shape === 'circle'
+            ? <Circle key={id} color={square.color} /> : 0))}
```
Comment (kaz231) [inline src/components/SquareBox/SquareBox.tsx]:
```suggestion
        {squares.map((square: Shape) => square.shape === 'square'
          ? <Square key={id} color={square.color} /> : <Circle key={id} color={square.color} />)}
```

We have to options so we can check only one shape and use the other as default.

Use should use `id` from line 37, not the index from map function.

File: src/components/SquareBox/SquareBox.tsx [hunk 3]
```diff
@@ -1,57 +1,75 @@
...
-    const [squares, setSquares] = React.useState<Array<string>>([]);
-
-    const deleteSquare = function () {
-        squares.pop();
-        setSquares(squares => [...squares]);
-    };
+  interface Shape {
+    color: string;
+    shape: string;
+  }
```
Comment (kaz231) [inline src/components/SquareBox/SquareBox.tsx]: Move this interface outside of SquareBox function, the body of SquareBox is re-run every time when the view has to be refreshed, so this interface is reacted over and over again.

File: src/components/SquareBox/SquareBox.tsx [hunk 4]
```diff
@@ -1,57 +1,75 @@
...
+  const id = React.useId();
+  const [squares, setSquares] = React.useState<Array<Shape>>([]);
+  const addCircle = () => {
+    setSquares((squares) => ([
```
Comment (kaz231) [inline src/components/SquareBox/SquareBox.tsx]: squares is already declared in 38, you should rather alias it

```
setSquares((prevSquares) => ...)
```

**Review by kaz231 — CHANGES_REQUESTED** [review]

File: src/components/SquareBox/SquareBox.tsx [hunk 5]
```diff
@@ -28,24 +27,23 @@ const styles = StyleSheet.create({
...
-    setSquares((squares) => ([
+    setSquares(() => ([
       ...squares, { color: randomElement(Object.values(COLORS)), shape: 'circle' },
     ]
     ));
   };
 
   const deleteSquare = () => {
     squares.pop();
-    setSquares((squares) => [...squares]);
```
Comment (oDoLaReKo) [inline src/components/SquareBox/SquareBox.tsx]: in this and in line 38 squares are not required but what to do with line 50 how should i change it ?

**Pull request merged**